src/coreclr/jit/llvm.cpp

```diff
@@ -1896,7 +1857,277 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
+
+        putArg->gtCorInfoType = CORINFO_TYPE_PTR;
+        lastArg               = _compiler->gtInsertNewCallArgAfter(putArg, insertAfterArg);
+
+        GenTree* returnAddrLclAfterCall = _compiler->gtNewLclvNode(returnTempNum, TYP_I_IMPL);
+        GenTree* indirNode;
+        if (callReturnType == TYP_STRUCT)
+        {
+            putArg->gtClsHnd                = calleeSigInfo.retTypeClass;
+            indirNode                       = new (_compiler, GT_OBJ) GenTreeObj(callReturnType, returnAddrLclAfterCall,
+                                                           _compiler->typGetObjLayout(calleeSigInfo.retTypeClass));
+            indirNode->AsBlk()->gtBlkOpKind = GenTreeBlk::BlkOpKindInvalid;
+        }
+        else
+        {
+            putArg->gtClsHnd = nullptr; // i8* for everything other than structs
+            indirNode        = _compiler->gtNewOperNode(GT_IND, callReturnType, returnAddrLclAfterCall);
+        }
+        indirNode->gtFlags |= GTF_IND_TGT_NOT_HEAP; // No RhpAssignRef required
+        LIR::Use callUse;
+        if (CurrentRange().TryGetUse(callNode, &callUse))
+        {
+            callUse.ReplaceWith(_compiler, indirNode);
+        }
+        else
+        {
+            callNode->ClearUnusedValue();
+        }
+
+        callNode->gtReturnType = TYP_VOID;
+        callNode->ChangeType(TYP_VOID);
+
+        CurrentRange().InsertBefore(callNode, shadowStackVar, offset, returnValueAddress, addrStore);
+        CurrentRange().InsertAfter(addrStore, returnAddrLcl, putArg);
+        CurrentRange().InsertAfter(callNode, returnAddrLclAfterCall, indirNode);
+    }
+
+    return lastArg;
+}
+
+void Llvm::failUnsupportedCalls(GenTreeCall* callNode, CORINFO_SIG_INFO &calleeSigInfo)
+{
+    // we can't do these yet
+    if (callNode->gtCallType == CT_INDIRECT || _isRuntimeImport(_thisPtr, callNode->gtCallMethHnd))
+    {
+        failFunctionCompilation();
+    }
+
+    // TODO-LLVM: not attempting to compile generic signatures with context arg via clrjit yet
+    if (calleeSigInfo.hasTypeArg())
+    {
+        failFunctionCompilation();
+    }
+
+    // this call has 3 args, but only the struct RuntimeTypeHandle in the siginfo
+    //             RuntimeTypeHandle handle = instance.GetInterfaceImplementation(new RuntimeTypeHandle(new
+    //             EETypePtr(interfaceType)));
+    //
+    // / --*t2 ref this in rcx
+    // + --*t19 int arg2 in rdx
+    // +--*t137 int arg1 in r10                  N007(27, 18)[000020]-- CXG-- -----t20 =
+    // *CALLV stub struct System.Runtime.InteropServices.IDynamicInterfaceCastable.GetInterfaceImplementation
+    if (callNode->IsVirtual())
+    {
+        // TODO-LLVM: what is the 3rd argument in this example and how can we get its type?
+        // looks like an interface call that in IL would be done with a call to FindInterfaceMethodImplementationTarget
```

Comment:
The `stub` of `CALLV stub struct` tells me this is a VSD dispatch and the `t137` is the address hidden argument VSD uses. So it is always `TYP_I_IMPL`.

Reply:
Thanks VSD - Virtual ... Dispatch ?

Reply:
[Virtual Stub Dispatch](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/virtual-stub-dispatch.md).